docs/book/component-guide/experiment-trackers/wandb.md

```diff
@@ -196,13 +197,109 @@ def my_step(
+
+You can enable or disable Weave for specific steps in your pipeline by configuring the `enable_weave` parameter in the `WandbExperimentTrackerSettings` (or setting it when registering the experiment tracker component):
+
+```python
+import weave
+from openai import OpenAI
+
+from zenml import pipeline, step
+from zenml.integrations.wandb.flavors.wandb_experiment_tracker_flavor import (
+    WandbExperimentTrackerSettings,
+)
+
+# Settings to enable Weave
+wandb_with_weave_settings = WandbExperimentTrackerSettings(
+    tags=["weave_enabled"],
+    enable_weave=True,  # Enable Weave integration
+)
+
+# Settings to disable Weave
+wandb_without_weave_settings = WandbExperimentTrackerSettings(
+    tags=["weave_disabled"],
+    enable_weave=False,  # Explicitly disable Weave integration
+)
+```
+
+#### Using Weave with ZenML Steps
+
+To use Weave with your ZenML steps, you need to:
+
+1. Configure your `WandbExperimentTrackerSettings` with `enable_weave=True`
+2. Apply the `@weave.op()` decorator to your step function
+3. Configure your step to use the Weights & Biases experiment tracker with your Weave settings
+
+Here's an example:
+
+```python
+@step(
+    experiment_tracker="wandb_weave",  # Your W&B experiment tracker component name
+    settings={"experiment_tracker": wandb_with_weave_settings},
+)
+@weave.op()  # The Weave decorator
+def my_step_with_weave() -> str:
+    """This step will use Weave for enhanced visualization"""
+    # Your step implementation
+    return "Step with Weave enabled"
+```
+
+{% hint style="warning" %}
+**Important**: The decorator order is critical. The `@weave.op()` decorator must be applied AFTER the `@step` decorator (i.e., closer to the function definition). If you reverse the order, your step won't work correctly.
+
+```python
+# CORRECT ORDER
+@step(experiment_tracker="wandb_weave")
+@weave.op()
+def correct_order_step():
+    ...
+
+# INCORRECT ORDER - will cause issues
+@weave.op()
+@step(experiment_tracker="wandb_weave")
+def incorrect_order_step():
+    ...
+```
+{% endhint %}
+
+To explicitly disable Weave for specific steps, while keeping the ability to use the `@weave.op()` decorator:
+
+```python
+@step(
+    experiment_tracker="wandb_weave",
+    settings={"experiment_tracker": wandb_without_weave_settings},
+)
+@weave.op()
+def my_step_without_weave() -> str:
+    """This step will not use Weave even with the @weave.op() decorator"""
+    # Your step implementation
+    return "Step with Weave disabled"
+```
+
+#### Weave Initialization Behavior
+
+When using Weave with ZenML, there are a few important behaviors to understand:
+
+1. If `enable_weave=True` and a `project_name` is specified in your W&B experiment tracker, Weave will be initialized with that project name.
+2. If `enable_weave=True` but no `project_name` is specified, Weave initialization will be skipped.
+3. If `enable_weave=False` (explicit disabling), Weave will be disabled with `settings={"disabled": True}`.
```

Comment:
This also requires a project name to work (Not sure why but at least in code that's what's needed right now)